sdk/keyvault/azure-keyvault-keys/tests/test_key_client.py

```diff
@@ -134,13 +135,15 @@ def _to_bytes(hex):
         self._validate_rsa_key_bundle(imported_key, client.vault_url, name, key.kty, key.key_ops)
         return imported_key
 
-    @parameterized.expand([param(is_hsm=b) for b in [True, False]], name_func=suffixed_test_name)
+    @parameterized.expand(
+        [(param(api_version=p[0], is_hsm=p[1])) for p in PARAMETER_COMBINATIONS], name_func=suffixed_test_name
+    )
     def test_key_crud_operations(self, **kwargs):
         is_hsm = kwargs.pop("is_hsm")
-        self._skip_if_not_configured(is_hsm)
+        self._skip_if_not_configured(kwargs.get("api_version"), is_hsm)
         endpoint_url = self.managed_hsm_url if is_hsm else self.vault_url
 
-        client = self.create_key_client(endpoint_url)
+        client = self.create_key_client(endpoint_url, **kwargs)
```

Comment:
When I look at this I still see boilerplate crying out for a decorator. Maybe I just don't see the blocker, does something prevent consolidating client creation into a decorator?

Reply:
`parameterized.expand` sends the `api_version` and `is_hsm` kwargs directly to the test method, so I don't think a decorator could take these into account when creating a client

Reply:
You mean, it sends those arguments directly to the function it decorates 😉 

```py
import functools
from parameterized import parameterized, param

def boilerplate(fn):
    @functools.wraps(fn)
    def wrapper(test_class_instance, api_version, is_hsm, **kwargs):
        test_class_instance._skip_if_not_configured(api_version, is_hsm)
        endpoint_url = test_class_instance.managed_hsm_url if is_hsm else test_class_instance.vault_url
        client = test_class_instance.create_key_client(endpoint_url, **kwargs)
        fn(client)
    return wrapper

class Tests(KeysTestCase, KeyVaultTestCase):
    @parameterized.expand([param(api_version=42, is_hsm=b) for b in (True, False)])
    @boilerplate
    def test_foo(self, client):
        ...
```

Reply:
😳